From 8dae99db0c2804415205f4d935db1ca09712e136 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 29 Dec 2022 14:23:21 -0800
Subject: [PATCH] Fix display too small issue for cgdisk

This patch allows the cgdisk utility to continue when a small display is
detected after flashing a warning about display issues.

This fixes an issue in the Clear Linux installer on older machines with
lower-resolution diplays (Lenovo ThinkPad S230u) where cgdisk reported the
screen was too small to continue and the user was unable to manually
partition their drive without switching TTYs.

On the mentioned machine this patch fixes the issue, and there are no
breaking issues in the cgdisk utility.
---
 gptcurses.cc | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/gptcurses.cc b/gptcurses.cc
index 80817e9..1fe001d 100644
--- a/gptcurses.cc
+++ b/gptcurses.cc
@@ -762,17 +762,15 @@ void GPTDataCurses::DrawMenu(void) {
 } // DrawMenu
 
 int GPTDataCurses::MainMenu(void) {
-   if (((LINES - RESERVED_TOP - RESERVED_BOTTOM) < 2) || (COLS < 80)) {
-      Report("Display is too small; it must be at least 80 x 14 characters!");
-   } else {
-      if (GPTData::Verify() > 0)
-         Report("Warning! Problems found on disk! Use the Verify function to learn more.\n"
-                "Using gdisk or some other program may be necessary to repair the problems.");
-      IdentifySpaces();
-      currentSpaceNum = 0;
-      DrawMenu();
-      AcceptInput();
-   } // if/else
+   if (((LINES - RESERVED_TOP - RESERVED_BOTTOM) < 2) || (COLS < 80))
+      Report("Display is less than 80 x 14 characters, this may cause display issues.");
+   if (GPTData::Verify() > 0)
+      Report("Warning! Problems found on disk! Use the Verify function to learn more.\n"
+             "Using gdisk or some other program may be necessary to repair the problems.");
+   IdentifySpaces();
+   currentSpaceNum = 0;
+   DrawMenu();
+   AcceptInput();
    endwin();
    return 0;
 } // GPTDataCurses::MainMenu
-- 
2.38.1

